### ISSUE TITLE
Can't run linter govet: can't typecheck package: can't init std types: import of type fmt.Stringer failed: can't find import: fmt

### ISSUE THREAD

**Issue by mbana** [golangci/golangci-lint#323]
1. I cannot get this tool to display the errors in VSCode. I just keep seeing these errors in the logs:

```sh
$ golangci-lint run --print-issued-lines=false --fast   
WARN [runner] Can't run linter govet: can't typecheck package: can't init std types: import of type fmt.Stringer failed: can't find import: fmt 
```

2. What is adding `--print-issued-lines=false` to the command line invocation? It's not present in my config.

---

Please include the following information:
> 1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)

```sh
golangci-lint --version
golangci-lint has version 1.11 built from 17508ab on 2018-10-28T15:21:50Z
```

> 2. Config file: `cat .golangci.yml`

```sh
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

> 3. Go environment: `go version && go env`

```sh
$ go version && go env
go version go1.11 darwin/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/mbana/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/mbana/home/mbana/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/opt/go/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/opt/go/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/mbana/work/openbankinguk/bitbucket/conformance-suite/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/kv/g0dcmzwx6kj367r3vgqmwk2m0000gp/T/go-build502429057=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

> 4. Verbose output of running: `golangci-lint run -v`

```sh
golangci-lint run --print-issued-lines=false --fast -v
INFO [config_reader] Config search paths: [./ /Users/mbana/work/openbankinguk/bitbucket/conformance-suite /Users/mbana/work/openbankinguk/bitbucket /Users/mbana/work/openbankinguk /Users/mbana/work /Users/mbana /Users /] 
INFO [lintersdb] Active 7 linters: [deadcode errcheck govet ineffassign structcheck typecheck varcheck] 
INFO [loader] Go packages loading at mode load types and syntax took 1.381573308s 
INFO [loader/astcache] Parsed AST of all pkg.GoFiles: [] for 258ns 
INFO [runner/skip dirs] sorted abs args: [/Users/mbana/work/openbankinguk/bitbucket/conformance-suite] 
INFO [runner] worker.3 took 5.149µs               
INFO [runner] worker.4 took 4.627µs               
INFO [runner] worker.5 took 560.455µs with stages: govet: 547.741µs 
WARN [runner] Can't run linter govet: can't typecheck package: can't init std types: import of type fmt.Stringer failed: can't find import: fmt 
INFO [runner] worker.1 took 1.479658ms with stages: errcheck: 1.469557ms 
INFO [runner] worker.8 took 2.295126ms with stages: structcheck: 2.284168ms, typecheck: 2.604µs 
pkg/model/match_test.go:65:16: Error return value of `json.Unmarshal` is not checked (errcheck)
pkg/model/match_test.go:75:16: Error return value of `json.Unmarshal` is not checked (errcheck)
pkg/model/specification_test.go:40:33: Error return value of `loadDefaultSpecifications` is not checked (errcheck)
pkg/server/server_test.go:354:16: Error return value of `json.Unmarshal` is not checked (errcheck)
INFO [runner] worker.6 took 3.795757ms with stages: deadcode: 3.785235ms 
INFO [runner] worker.2 took 4.181231ms with stages: varcheck: 4.173949ms 
pkg/model/match.go:318:6: `checkUnimplemented` is unused (deadcode)
pkg/generation/generation_test.go:62:6: `printSpec` is unused (deadcode)
pkg/generation/generation.go:108:6: `printImplemented` is unused (deadcode)
pkg/generation/generation.go:128:6: `printResourceIds` is unused (deadcode)
INFO [runner] worker.7 took 11.728872ms with stages: ineffassign: 11.716562ms 
INFO [runner] Workers idle times: #1: 10.215812ms, #2: 7.537581ms, #3: 11.652166ms, #4: 11.628215ms, #5: 11.206873ms, #6: 7.900183ms, #8: 9.404207ms 
INFO [runner/max_same_issues] 4/7 issues with text "ineffectual assignment to `err`" were hidden, use --max-same-issues 
pkg/model/model.go:76:2: ineffectual assignment to `req` (ineffassign)
INFO [runner] processing took 3.175638ms with stages: exclude: 846.258µs, source_code: 636.62µs, path_prettifier: 500.121µs, autogenerated_exclude: 472.53µs, skip_dirs: 332.463µs, cgo: 193.412µs, nolint: 137.217µs, max_same_issues: 37.822µs, uniq_by_line: 7.228µs, path_shortener: 4.016µs, max_from_linter: 3.447µs, max_per_file_from_linter: 3.041µs, diff: 831ns, skip_files: 632ns 
pkg/model/model.go:149:2: ineffectual assignment to `err` (ineffassign)
internal/pkg/version/version.go:140:17: ineffectual assignment to `err` (ineffassign)
internal/pkg/version/version.go:141:18: ineffectual assignment to `err` (ineffassign)
internal/pkg/version/version_test.go:25:2: ineffectual assignment to `flag` (ineffassign)
pkg/model/model_test.go:170:2: ineffectual assignment to `result` (ineffassign)
INFO Memory: 15 samples, avg is 68.8MB, max is 68.8MB 
INFO Execution took 1.495711752s                  
```

---

**Comment by mbana** [comment]
Any ideas?

**Comment by slimsag** [comment]
This also appears to happen when just running golangci-lint on WASM code:

```bash
$ GOOS=js GOARCH=wasm golangci-lint run
WARN [runner] Can't run linter govet: can't typecheck package: can't init std types: import of type fmt.Stringer failed: can't find import: fmt 
```

**Comment by ww-daniel-mora** [comment]
I am also running into this issue after switching my repo to use modules. Other people in the same code base don't seem to be seeing it though.

**Comment by dgtony** [comment]
I've encountered this issue using golangci-lint releases for the wrong arch: e.g. run i386 version on amd64.

**Comment by ww-daniel-mora** [comment]
@jirfag I may have diagnosed the issue. Machines that have the standard go install according to the go instructions https://golang.org/doc/install#install with go installed at /usr/local/go were not experiencing the issue. Machines that had only used a brew installation of go were experiencing the issue. I suspect something somewhere had this path hard coded. Upgrading to the newest version of go 1.12 has resolved the issue for multiple people on my team. So something may have been fixed in the upgrade. I have not tested to see if doing a standard install would also have fixed it.

**Comment by jaloren** [comment]
I can easily reproduce the problem with go 1.12.1, when doing cross compilation. I do not have multiple installs golang on my mac and I only installed using the official installer. I did not homebrew install.

Steps to reproduce:

1. make sure you are on a mac.
2. cd to a directory go source code that is also a go module.
3. make sure go build is module aware (e.g set the GO111MODULE to on or make sure this directory is outside the go path)
4. export GOOS=linux
5. golangci-lint run ./...

If unset the GOOS environment variable, the warning no longer displays.

**Comment by jaloren** [comment]
I compiled the latest version of golangci-lint today.

golangci-lint has version v1.15.0-21-gf516771 built from f516771 on Sun Mar 24 07:34:52 CDT 2019

The issue no longer reproduces.

**Comment by jirfag** [comment]
can anyone reproduce it now in a normal (no `GOOS=linux` and invalid arch) situation?

**Comment by mbana** [comment]
Feel free to close. If it comes up again, I'll re-open with a reproduction minimal reproduction repository.
Thanks for looking into it.
